### PR TITLE
Adds a new event, Queen Mother's fireproofing

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

It's a new event, Tivi wants more of them and I sat down and made some.

Queen Mother's fireproofing acts like a team wide application of hivelord gel, it's low probability and limited to two times. It cannot activate unless the xeno team has enough players for a Queen.

## Why It's Good For The Game

More events good.

## Changelog
:cl:
add: Added a new event, Queen Mother fireproofing.
/:cl:

### CODE REVIEW

**Comment by Autisem** [comment]
how long does this last? it feels like this would happen alot when its not needed and wear off when it is needed

File: code/modules/events/queen_mothers_fireproof.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	gamemode_blacklist = list("Combat Patrol","Civil War","Sensor Capture")
+
+/datum/round_event_control/queen_mothers_fireproof/can_spawn_event(players_amt, gamemode)
+	if(length(GLOB.alive_xeno_list) <= 8)
+		return FALSE
+	return ..()
+
+/datum/round_event/queen_mothers_fireproof/start()
+	xeno_message("The Queen Mother has temporarily reinforced our armor with fireproof coating, strike before it wears off!")
+	for(var/mob/living/carbon/xenomorph/X in GLOB.alive_xeno_list)
```
Comment (Lumipharon) [inline code/modules/events/queen_mothers_fireproof.dm:17]: Don't use X, use proper var names.

File: code/modules/events/queen_mothers_fireproof.dm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		if(isminion(X))
+			continue
+		fireproof_xeno(X)
+
+/datum/round_event/queen_mothers_fireproof/proc/fireproof_xeno(mob/living/carbon/xenomorph/X)
```
Comment (Lumipharon) [inline code/modules/events/queen_mothers_fireproof.dm:22]: var names.

File: code/modules/events/queen_mothers_fireproof.dm [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	X.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
+	var/sound/queen_sound = sound(get_sfx("queen"), channel = CHANNEL_ANNOUNCEMENTS)
```
Comment (Lumipharon) [inline code/modules/events/queen_mothers_fireproof.dm:24]:
```suggestion
	var/sound/queen_sound = sound(get_sfx("queen"), channel = CHANNEL_ANNOUNCEMENTS, volume = 50)
```
default volume is 100, 50 is what's used for queen mother announcements

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by RipGrayson** [comment]
I've decided I don't like this event, it doesn't leave a lasting impact on the round besides the 60 seconds it's active.

**Pull request closed without merging**